src/server/game/Entities/Item/Item.cpp

```diff
@@ -91,6 +91,7 @@ void AddItemsSetItem(Player* player, Item* item)
                 }
 
                 // spell casted only if fit form requirement, in other case will casted at form change
+                if (sScriptMgr->CanItemApplyEquipSpell(player, item))
```

Comment:
it's not clear what this `if` is controlling. So please use brackets and indentation